src/st/no0/57C20.c

```diff
@@ -51,7 +51,538 @@ void func_us_801D91C4(Entity* self) {
+
+            self->posX.i.hi = 0x20;
+            self->posY.i.hi = posY - 0x28;
+
+            self->palette = *D_us_80180BDE;
+            self->flags &= ~FLAG_POS_CAMERA_LOCKED;
+            self->step_s++;
+            break;
+        case 1:
+            prim = self->ext.prim;
+            self->animCurFrame = 0;
+            prim->type = PRIM_ENV;
+
+            dr_env = LOW(prim->r1);
+            drawEnv = g_CurrentBuffer->draw;
+            drawEnv.isbg = 0;
+            drawEnv.dtd = 0;
+
+            posY = self->params ? 0x180 : 0x100;
+
+            clipRect.x = 0;
+            clipRect.y = posY;
+            clipRect.w = 0x80;
+            clipRect.h = 0x80;
+            drawEnv.clip = clipRect;
+            drawEnv.ofs[0] = 0;
+            drawEnv.ofs[1] = 0x100;
+            SetDrawEnv(dr_env, &drawEnv);
+
+            prim->priority = 0xF;
+            if (self->params) {
+                prim->priority += 4;
+            }
+
+            prim->drawMode = DRAW_DEFAULT;
+            self->ext.et_801D9264.y = 0x28;
+            self->ext.et_801D9264.timer = 0x10;
+            self->step_s++;
+            /* fallthrough */
+        case 2:
+            if (!(g_Timer & 7)) {
+                PlaySfxPositional(0x65B);
+            }
+            prim = self->ext.et_801D9264.unkA4;
+            posX = Random() & 0x3F;
+            posY = self->ext.et_801D9264.y;
+            if (!(g_Timer & 0xF)) {
+                newEntity = AllocEntity(&D_800762D8, (Entity*)&g_EvHwCardEnd);
+                if (newEntity != NULL) {
+                    CreateEntityFromCurrentEntity(2, newEntity);
+                    newEntity->posX.i.hi = prim->x0 + posX;
+                    newEntity->posY.i.hi = prim->y2 + posY - 0x30;
+                    newEntity->params = 3;
+                }
+            }
+            newEntity = AllocEntity(&D_800762D8, (Entity*)&g_EvHwCardEnd);
+            if (newEntity != NULL) {
+                CreateEntityFromCurrentEntity(0x46, newEntity);
+                newEntity->posX.i.hi = (self->posX.i.hi - 0x20) + posX;
+                newEntity->posY.i.hi = (self->posY.i.hi + posY) + 4;
+                newEntity->params = 1;
+                newEntity->facingLeft = colRet;
+                newEntity->zPriority = 0x10;
+                if (self->params) {
+                    newEntity->zPriority += 4;
+                }
+            }
+            if (!--self->ext.et_801D9264.timer) {
+                self->ext.et_801D9264.timer = 2;
+                self->ext.et_801D9264.y -= 2;
+                if (self->ext.et_801D9264.y < -0x28) {
+                    self->ext.et_801D9264.timer = 0x40;
+                    self->step_s++;
+                }
+            }
+            break;
+        case 3:
+            if (!--self->ext.et_801D9264.timer) {
+                DestroyEntity(self);
+                return;
+            }
+        }
+        break;
+    case 255:
+        FntPrint("charal %x\n", self->animCurFrame);
```

Comment:
Should be able to replace this case with an include of https://github.com/Xeeynamo/sotn-decomp/blob/master/src/st/pad2_anim_debug.h

See https://github.com/Xeeynamo/sotn-decomp/blob/2765a6eb80a85e123952b2468523f12310e7b395/src/st/no0/e_ouija_table.c#L147 for example